core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -981,7 +981,13 @@ private[spark] object Utils extends Logging {
       return cached
     }
 
-    val indx: Int = hostPort.lastIndexOf(':')
+    val indx: Int =
+      // Interpret hostPort as literal IPv6 address if it contains two ore more colons
+      // scalastyle:off SingleSpaceBetweenRParenAndLCurlyBrace
+      if (hostPort.matches("(([0-9a-f]*):([0-9a-f]*)){2,}")) -1
```

Comment:
Hex digits can be uppercase right?
Should the pattern it not be more like `[0-9a-f]*(:[0-9a-f]*)+` match a number, then colon-number colon-number pairs, not number-colon-number number-colon-number sequences?
It might end up being equivalent because the match is for 0 or more digits.

This allows some strings that it shouldn't like "::::", but, the purpose isn't to catch every possible case I guess. It would fail name resolution.

I thought Inet6Address would just provide parsing for this but I guess not.

Reply:
Yes a real parser would be much better!! 
I hope the methods will check the input. Like the name resolver... 

At this point I thought more about the separation of the port.
I think it is important to check if two colons exists, otherwise this expression accepts hostnames like abc:123